### PR TITLE
[dynamo] support tensor.mT

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #91322
* __->__ #91262
* #91218

Special handling for tensor.mT to avoid graph breaks.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by yanboliang — APPROVED** [review]

File: torch/_dynamo/variables/tensor.py [hunk 1]
```diff
@@ -153,8 +153,19 @@ def var_getattr(self, tx, name):
...
+                args = [
+                    variables.ConstantVariable(i) for i in range(self.ndim - 1, -1, -1)
+                ]
+                result = self.call_method(tx, "permute", args, {})
+        elif name == "mT" and self.ndim != 1:
+            if self.ndim == 0:
+                result = self
+            else:
+                args = [variables.ConstantVariable(i) for i in [-2, -1]]
+                result = self.call_method(tx, "transpose", args, {})
```
Comment (ezyang) [inline torch/_dynamo/variables/tensor.py:168]: So what ever happened to my original comment about using __get__

Comment (davidberard98) [reply torch/_dynamo/variables/tensor.py:168]: sorry, don't think I saw that comment?

Comment (ezyang) [reply torch/_dynamo/variables/tensor.py:168]: The way this PR is constructed is you are manually desugaring the mH attribute into am equivalent series of transpose calls. The main problem with doing it this way is it's easy to do it wrong (dual source of truth) and the resulting program doesn't really capture what the previous program did. Ideally we would trace an mH access directly into the FX graph, however FX only supports function calls. The proposal was to make a function call with the getter function Tensor.mH.__get__. I vaguely remember someone telling me this didn't work but I'd quite like to know why it didn't work.

**Comment by davidberard98** [comment]
closing because this is fixed in https://github.com/pytorch/pytorch/pull/90463, plus Ed's comments about preferring a more general approach.

**Pull request closed without merging**